Makefile

```diff
@@ -25,14 +25,17 @@ build:
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -a -installsuffix cgo -ldflags '-s' -o $(BINDIR)/boot boot.go || exit 1
 
 docker-build:
+	# build the minio server
 	docker build -t minio mc
 	docker cp `docker run -d minio`:/go/bin/minio $(BINDIR)
+
+	# build the main image
 	docker build --rm -t ${IMAGE} rootfs
 	# These are both YAML specific
 	perl -pi -e "s|image: [a-z0-9.:]+\/deis\/${SHORT_NAME}:[0-9a-z-.]+|image: ${IMAGE}|g" ${RC}
 	perl -pi -e "s|release: [a-zA-Z0-9.+_-]+|release: ${VERSION}|g" ${RC}
 
-docker-push:
+docker-push: docker-build
```

Comment:
thanks @bacongobbler, made the change at https://github.com/deis/minio/pull/16

also, today I forgot that you could run two build targets in succession with `make docker-build docker-push` until I revisited this PR discussion. some days my make-fu is that bad...